python/dgl/graphbolt/feature_store.py

```diff
@@ -26,6 +26,46 @@ def read(self, ids: torch.Tensor = None):
         """
         raise NotImplementedError
 
+    def read_async(self, ids: torch.Tensor):
+        """Read the feature by index asynchronously.
+        Parameters
+        ----------
+        ids : torch.Tensor
+            The index of the feature. Only the specified indices of the
+            feature are read.
+        Returns
+        -------
+        A generator object.
+            The returned generator object returns a future on
+            `read_async_num_stages(ids.device)`th invocation. The return result
+            can be accessed by calling `.wait()`. on the returned future object.
+            It is undefined behavior to call `.wait()` more than once.
+        Example Usage
+        --------
+        >>> import dgl.graphbolt as gb
+        >>> feature = gb.Feature(...)
+        >>> ids = torch.tensor([0, 2])
+        >>> async_handle = feature.read_async(ids)
+        >>> for _ in range(feature.read_async_num_stages(ids.device)):
+        ...     future = next(async_handle)
+        >>> result = future.wait()  # result contains the read values.
+        """
+        raise NotImplementedError
+
+    def read_async_num_stages(self, ids_device: torch.device):
```

Comment:
If you already returned an async_handle, why do you need to define read_async_num_stages?
Consider merging?

Reply:
Because `read_async` returns a generator object. There is no way to also return how many stages are in it.

Reply:
e.g. implement the len function of the object?

Reply:
How can I do that?